diesel/src/sqlite/connection/raw.rs

```diff
@@ -194,12 +251,150 @@ extern "C" fn run_custom_function<F>(
     }
 }
 
+// Need a custom option type here, because the std lib one does not have guarantees about the discriminate values?
```

Comment:
Ok I had a short chat with Ralf Jung in zulip. He says the following about the general approach:

> so given that RFC, I agree that it is okay to transmute a "bunch of zero bytes" to MyOption.
however, notice that MyOption contains padding, so if you transmute it back those padding bytes will be uninitialized.

so it seems to be fine what we are doing here.

Additionally I've found a section in the [corresponding RFC](https://github.com/rust-lang/rfcs/blob/master/text/2195-really-tagged-unions.md#guide-level-explanation) that specifically answers the question raised in this comment.

> It relies on the fact that the Some-ness of an Option (or the tag of any repr(Rust) enum) can't rely on the tag of a repr(C/Int) enum. Or in other words, repr(C/Int) enums have opaque tags. The cost of making this work is that Option<MyEnum> would have to be larger than MyEnum.

> It would be nice for this to work, but if you really need it, you can just define #[repr(u8)] COption<T> { ... } and use that.